src/web/js/trove/world-lib.js

```diff
@@ -750,10 +759,20 @@
                 k2();
               } else {
                 if (extras.closeWhenStop) {
-                  if (extras.closeBigBangWindow) {
-                    extras.closeBigBangWindow();
+                  if (lif) {
+                    runtime.ffi.throwMessageException('Cannot have both last-image and close-when-stop(true)');
+                  } else {
+                    if (extras.closeBigBangWindow) {
+                      extras.closeBigBangWindow();
+                    }
+                    Jsworld.shutdownSingle({cleanShutdown: true});
                   }
-                  Jsworld.shutdownSingle({cleanShutdown: true});
+                } else if (lif) {
+                  var handler = lif();
```

Comment:
The example should work as is. Paste in entirety into the def window, and click Run.